packages/cms/src/lokalize/add-text.ts

```diff
@@ -1,120 +1,150 @@
+
+        console.log(`Successfully created ${textDocument.key}`);
+
+        existingKeys.push(textDocument.key);
+
+        additionsCounter++;
+      }
+    }
+    {
+      const response = await prompts([
         {
           type: 'confirm',
           name: 'continue',
-          message: 'Create another text in the same subject?',
-        },
-      ],
-      {
-        onCancel: () => {
-          continueCreating = false;
+          message: 'Add another text?',
+          onState,
         },
+      ]);
+
+      if (!response.continue) {
+        if (additionsCounter > 0) {
+          console.log('Updating text export...');
+          await exportLokalizeTexts();
+        }
+        break;
       }
-    );
-
-    if (response.confirmed) {
-      await client.create(textDocument);
-
-      const key = `${textDocument.subject}.${textDocument.path}`;
-      await appendTextMutation('add', key);
-
-      console.log(`Successfully created ${key}`);
-    }
-
-    if (!response.continue) {
-      continueCreating = false;
     }
   }
 })().catch((err) => {
   console.error('An error occurred:', err.message);
   process.exit(1);
 });
 
-async function createTextDocumentForSubject(
-  subject: string,
-  allTexts: LokalizeText[]
-) {
+/**
+ * There is currently no native way to exit prompts on ctrl-c. This is a
+ * workaround that needs to be added to every prompts instance. For more info
+ * see: https://github.com/terkelg/prompts/issues/252#issuecomment-778683666
+ */
+function onState(state: { aborted: boolean }) {
+  if (state.aborted) {
+    if (additionsCounter > 0) {
+      console.log('Updating text export...');
+      exportLokalizeTexts().finally(() => {
+        process.exit(0);
+      });
+    } else {
+      process.nextTick(() => {
+        process.exit(0);
+      });
+    }
+  }
+}
+
+async function createTextDocument(existingKeys: string[], initialKey?: string) {
   const response = await prompts([
     {
       type: 'text',
-      name: 'path',
-      message: 'What is the path in dot notation?',
-      format: (x: string) => x.toLowerCase(),
-      /**
-       * Do not allow creating a text with a key that already exists
-       */
-      validate: (x: string) =>
-        allTexts.find((text) => text.key === `${subject}.${x}`) === undefined,
+      name: 'key',
+      message: 'What is the key? (Use snake_cased.dot.notation)',
+      initial: initialKey,
+      validate: (x: string) => {
+        /**
+         * Validation requires the key to be new and also to only contain
+         * lower-snake-case paths in dot notation.
+         *
+         * @TODO improve the regexp so that the key is not allowed to end in a
+         * dot. Also we should enforce at least 1 dot to have subject+path
+         * separation in Sanity.
+         * https://regex101.com/r/RJaXUv/1
+         */
+        return (
+          existingKeys.find((key) => key === x) === undefined &&
+          /^([a-z_]+\.?)+$/.test(x)
```

Comment:
```suggestion
          /^([a-z_\.]+[a-z_]$/.test(x)
```

and then you can remove the `todo`

Reply:
Nice, this works (without the parenthesis)